### ISSUE TITLE
cmake CMAKE_INSTALL_LIBDIR sometimes absolute path

### ISSUE THREAD

**Issue by mwinter-osr** [rtrlib/rtrlib#147]
Something I ran into, trying to get the RPM package working on CentOS 6. I have close to zero cmake experience, so need advise from someone:

If I build on CentOS 7 (and other newer systems):, the rtrlib.pc ends up like this:
```prefix=/usr
includedir=${prefix}/include
libdir=${prefix}/lib64

Name: rtrlib
Description: 
Version: 0.5.0
Requires: libssh >= 0.5.0
Libs: -L${libdir} -lrtr
Cflags: -I${includedir}
```

However on a CentOS 6 system, I get a wrong path for `libdir`:
```prefix=/usr
includedir=${prefix}/include
libdir=${prefix}//usr/lib64

Name: rtrlib
Description: 
Version: 0.5.0
Requires: libssh >= 0.5.0
Libs: -L${libdir} -lrtr
Cflags: -I${includedir}
```

This ends up with `libdir=/usr//usr/lib64` instead of `libdir=/usr/lib64`

The issue is based on `CMakeLists.txt`:
```
set (PKG_CONFIG_LIBDIR     "\${prefix}/${CMAKE_INSTALL_LIBDIR}")
```

However, `CMAKE_INSTALL_LIBDIR` is **NOT** always a relative path. Older systems seem to provide an absolute path.
See discussion at 
https://bugzilla.redhat.com/show_bug.cgi?id=1301720
and https://cmake.org/pipermail/cmake/2012-February/049250.html

Any thoughts on how to resolve this?

**Comment by waehlisch** [comment]
@fho any idea?

**Comment by fho** [comment]
@waehlisch no, sorry :-)

**Comment by waehlisch** [comment]
Hmm, I also don't have a good idea. It's about a workaround of an old bug. @smlng?

@mwinter-osr what do you prefer?

**Comment by mwinter-osr** [comment]
I've tried some workaround, but so far with no success. I'll try a bit more over the weekend.
But I'm a cmake novice (never used cmake before). I was counting on someone here understanding cmake better than I do.

**Comment by mroethke** [comment]
It seems the old cmake version is the problem.
I have found two possible solutions

1. Ship the cmake module GNUInstallDirs and include it if necessary. 
2. Just check if CMAKE_INSTALL_LIBDIR is absolute and handle it accordingly.

I did not check if the first one acualty works, the second one is implemented in #150

**Comment by waehlisch** [comment]
@mwinter-osr does PR #150 work for you?

**Comment by waehlisch** [comment]
btw, @mroethke, I like your second solution!

**Comment by mwinter-osr** [comment]
Thanks, PR #150 fixes the issue. Works great.

I'll update PR #141 after this is merged.

**Comment by waehlisch** [comment]
Fixed in PR #150.
